### PR TITLE
Add tree route and recursive option

### PR DESCRIPTION
**Pull request by billytrend**

### CODE REVIEW

File: src/dtos/git-file-object-content.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ import { ApiModelProperty } from "@nestjs/swagger";
 import { GitObjectContent } from "./git-object-content";
 
 export class GitFileObjectContent extends GitObjectContent {
-  @ApiModelProperty({ type: String })
-  public content: string;
+  @ApiModelProperty({ type: String, required: false })
```
Comment (timotheeguerin) [inline src/dtos/git-file-object-content.ts]: I think here its better to do have 2 class one with content one without. that way you don't have it potentially undefined even if you know it isn't

File: src/controllers/content/content.controller.ts [hunk 2]
```diff
@@ -13,15 +13,17 @@ export class ContentController {
...
   @ApiNotFoundResponse({})
   public async getContents(
     @Param("remote") remote: string,
     @Param("path") path: string | undefined,
     @Query("ref") ref: string | undefined,
+    @Query("recursive") recursive: string | undefined,
     @Auth() auth: RepoAuth,
   ) {
-    const content = await this.contentService.getContents(remote, path, ref, { auth });
+    const content = await this.contentService.getContents(remote, path, ref, Boolean(recursive), true, { auth });
```
Comment (timotheeguerin) [inline src/controllers/content/content.controller.ts]: is that a safe one here `Boolean(recursive)` what option does it cover?

Comment (timotheeguerin) [reply src/controllers/content/content.controller.ts]: I'm going to go with no
image

Comment (billytrend) [reply src/controllers/content/content.controller.ts]: Oo TIL so.. eval(recursive)? 😄

File: src/dtos/git-file-object-content.ts (hunk 1 above)
Comment (billytrend) [reply src/dtos/git-file-object-content.ts]: Yeah good point

File: src/controllers/content/content.controller.ts (hunk 2 above)
Comment (timotheeguerin) [reply src/controllers/content/content.controller.ts]: haha

File: src/dtos/git-file-object-content.ts (hunk 1 above)
Comment (timotheeguerin) [reply src/dtos/git-file-object-content.ts]: I did that for `GitCommit` and `GitCommitRef` for refernece

**Comment by timotheeguerin** [comment]
Make sure you update the version too so the sdk get published

File: src/utils/misc-utils.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 export const notUndefined = <T>(x: T | undefined): x is T => x !== undefined;
 export const delay = (timeout?: number) => new Promise(r => setTimeout(r, timeout));
+export const parseBooleanFromURLParam = (bool: string | undefined) => bool !== "false" && bool !== undefined;
```
Comment (timotheeguerin) [inline src/utils/misc-utils.ts]: would it not be better to have `=== "true"`?

Comment (billytrend) [reply src/utils/misc-utils.ts]: I was thinking about this.. sometimes "?recursive" is a valid way of saying true afaik

Comment (billytrend) [reply src/utils/misc-utils.ts]: so I guess I could do "true" or "" ?

**Pull request merged**